dbsnap_verify/__init__.py

```diff
@@ -0,0 +1,176 @@
+from rds_funcs import (
+    get_latest_snapshot,
+    get_database_description,
+    restore_from_latest_snapshot,
+    modify_db_instance_for_verify,
+    dbsnap_verify_db_id,
+    destroy_database,
+    destroy_database_subnet_group,
+    rds_event_messages,
+)
+
+from state_doc import (
+    current_state,
+    transition_state,
+    get_or_create_state_doc,
+)
+
+from time_funcs import (
+    tomorrow_timestamp,
+    now_datetime,
+    timestamp_to_datetime,
+    datetime_to_date_str,
+    three_days_prior,
+)
+
+import boto3
+
+from sys import stdout
+import logging
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+logger.addHandler(logging.StreamHandler(stdout))
+
+
+def wait(state_doc, rds_session):
+    min_timestamp = state_doc["snapshot_minimum_timestamp"]
+    min_datetime = timestamp_to_datetime(min_timestamp)
+    logger.info("Looking for a snapshot of {} older then {}".format(
+        state_doc["database"],
+        datetime_to_date_str(min_datetime)
+    ))
+    snapshot_desc = get_latest_snapshot(rds_session, state_doc["database"])
+    if snapshot_desc["SnapshotCreateTime"] >= min_datetime:
+        # if the latest snapshot is older then the minimum, restore it.
+        restore(state_doc, rds_session)
+    elif min_datetime >= three_days_prior(now_datetime()):
+        # continue wating for a new snapshot, to restore.
+        transition_state(state_doc, "wait")
+        logger.info("Did not find a snapshot of {} older then {}".format(
+            state_doc["database"],
+            datetime_to_date_str(min_datetime)
+        ))
+        logger.info("Going to sleep.")
+    else:
+        logger.warning("Alert! we never found a snapshot of {} older then {}".format(
+            state_doc["database"],
+            datetime_to_date_str(min_datetime)
+        ))
+        alarm("asdfasdfasdkfjnasdf naskdfn aksdjfn")
+
+
+def restore(state_doc, rds_session):
+    tmp_db_description = get_database_description(
+        rds_session, state_doc["tmp_database"]
+    )
+    if tmp_db_description is None:
+        logger.info(
+            "Restoring snapshot of {database} to {tmp_database}".format(
+                **state_doc
+            )
+        )
+        transition_state(state_doc, "restore")
+        sn_ids = state_doc["database_sn_ids"]
```

Comment:
I may be blanking here - what's an `sn_id`?

Reply:
`database_subnet_ids`